### PR TITLE
Add our current architecture to the blog

### LINKED ISSUE #1427: architecture diagram?
would be nice if there was some sort of diagram of all the moving parts (nothing crazy fancy or anything) just to help potential new contributors get up to speed on all the moving parts (fastapi, nextjs, redis, etc). 

not sure if anything like that exists in docs yet anywhere?

### PR DESCRIPTION
**Pull request by AbdBarho**
might be a good point to discuss our plan so far and even the future.

Closes #1427

Also fixed our yarn setup which had unmet peer dependencies.

link to rendered diagram: https://github.com/LAION-AI/Open-Assistant/blob/arch/docs/blog/2023-02-11-architecture.md

### CODE REVIEW

File: docs/blog/2023-02-11-architecture.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: The Architecture so Far!
+description: This is the architecture of our app so far.
+slug: 2023-02-11-architecture
+authors: AbdBarho
+tags: [open-assistant]
+---
+
+I thought it would make sense to document our architecture this as a blog post,
```
Comment (andrewm4894) [inline docs/blog/2023-02-11-architecture.md:9]:
```suggestion
I thought it would make sense to document our architecture as a blog post,
```

**Review by andrewm4894 — APPROVED** [review]

File: docs/blog/authors.yml [hunk 2]
```diff
@@ -3,3 +3,7 @@ yk:
   title: Project Lead
   url: https://www.ykilcher.com/
   image_url: https://www.ykilcher.com/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fheadshot.ff3a7ee3.webp&w=3840&q=75
+AbdBarho:
+  name: Abdullah Barhoum
+  title: Semi-professional Döner Enjoyer
```
Comment (andreaskoepf) [inline docs/blog/authors.yml:8]: :-) haha ...

**Review by andreaskoepf — APPROVED** [review]
OK :-) .. thanks for the blog post. The inference architecture might have to be updated a bit later .. bit as a first impression this works well.

**Comment by bitplane** [comment]
Cool, mermaid looks pretty good inline. I think I'm sold on this!

File: docs/blog/2023-02-11-architecture.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    direction TB
+    inferenceServer[Server] -.- infpostgres[(PostgreSQL)]
+  end
+
+  subgraph workers[Inference Workers]
+    direction TB
+    worker1[Worker] --- textgen1[Text\ngenerator]
+    worker2[Worker] --- textgen2[Text\ngenerator]
+    worker3[Worker] --- textgen3[Text\ngenerator]
+  end
```
Comment (pdxjohnny) [inline docs/blog/2023-02-11-architecture.md:41]: We've been playing with ActivityPub to enable multiple workers to provide feedback via `inReplyTo` and threads, mimicking human behavior. Wasn't sure where else to post so posting here. The hope is that our models can collectivly respond, and the user or users AI agent can sift through and find the responses that are most helpful to them within the context of the conversation. Something like ActivityPub based communication (Rapunzel, ATProto come to mind) would enable folks AI's to collaboratively provide their responeses.

- References
  - [WIP: IETF SCITT: Use Case: OpenSSF Metrics: activitypub extensions for security.txt](https://github.com/ietf-scitt/use-cases/blob/fd2828090482fe63a30a7ddd9e91bdb78892a01e/openssf_metrics.md#activitypub-extensions-for-securitytxt)

Cheers! 🥳

**Pull request merged**